Yank/restraints.py

```diff
@@ -714,22 +787,24 @@ def _create_restraint_force(self, particle1, particle2):
         parameter_names, parameter_values = zip(*self._bond_parameters.items())
 
         # Create bond force and lambda_restraints parameter to control it.
-        force = openmm.CustomBondForce('lambda_restraints * ' + self._energy_function)
-        force.addGlobalParameter('lambda_restraints', 1.0)
+        if len(particles1) == 1 and len(particles2) == 1:
+            # CustomCentroidBondForce works only on 64bit platforms. When the
+            # restrained groups only have 1 particle, we can use the standard
+            # CustomBondForce so that we can support 32bit platforms too.
+            energy_function = self._energy_function.replace('distance(g1,g2)', 'r')
+            force = openmm.CustomBondForce('lambda_restraints * ' + energy_function)
+            force.addBond(particles1[0], particles2[0], parameter_values)
+        else:
```

Comment:
OpenMM already raises a very clear exception if that's the case on `Context` creation.